Add CONTRIBUTING.md to solicit contributions (#846)
jsx-eslint/eslint-plugin-jsx-a11y: CONTRIBUTING.md

# Contributing to ``eslint-plugin-jsx-a11y``

Thank you for your interest in becoming a part of the ongoing development of ``eslint-plugin-jsx-a11y``. 

This guide covers our practices and recommendations for contributing to the project.

## Table of Contents

[Ways to contribute](#ways-to-contribute)

[Making your first contribution](#making-your-first-contribution)

[Pull requests](#pull-requests)

[Commit messages](#commit-messages)

[Code of Conduct](#code-of-conduct)

## Ways to contribute

There are various ways to get involved, here are a few suggestions of things we'd love some help with.

#### Resolving existing issues

>  You can consider helping out with [issues](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues) already requiring attention.

Reviewer: should this link to the "help wanted" label?

#### Reporting issues

> If you run into problems while using  ``eslint-plugin-jsx-a11y``, you can report them by opening a new issue  and providing as much context as you can about what you're running into along with a failing test or code sample to help others verify and ultimately fix the issue.

Reviewer: ```suggestion
If you run into problems while using `eslint-plugin-jsx-a11y`, you can report them by opening a new issue and providing as much context as you can about what you're running into, along with a failing test or code sample to help others verify and ultimately fix the issue.
```

> > **Note:** Before submitting an issue, please check open and closed issues to see if the problem has already been reported.

Reviewer: ```suggestion
> **Note:** Before filing an issue, please check both open and closed issues to see if the problem has already been reported.
```

> If you want to propose a new rule, please follow these [instructions](https://eslint.org/docs/developer-guide/contributing/new-rules).

Reviewer: these are eslint core's guidelines - probably worth folks reading, but not necessarily the criteria we'd apply here.

Author: thank you for pointing out, I removed the reference😄

#### Documentation

> We are happy to welcome contributions  from anyone willing to improve documentation by adding missing information or making it more consistent and coherent.

Reviewer: ```suggestion
We are happy to welcome contributions from anyone willing to improve documentation by adding missing information or making it more consistent and coherent.
```

## Making your first contribution

* [Fork the repo and clone your fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo)

* Create a new branch
  ```
  git checkout -b issue1234
  ```
* Get all dependencies
  ```
  npm install
  ```
* Run the tests and the build
  ```

>   npm t && npm run build
>   ```
> * After making changes, verify them and update your code if there’re any failing tests
>   ```
>   npm run test

Reviewer: ```suggestion
  npm test && npm run build
  ```
* After making changes, verify them and update your code if there’re any failing tests
  ```
  npm run tests-only
```

  ```
  
* Add changes
  ```
  git add .
  git commit
  ```
* Create a commit message that follows our [standards](#commit-messages)
* Push your changes
  ```
  git push origin issue1234
  ```
* Create a pull request and iterate over your solution 
	

## Pull requests

Create a new branch

```
git checkout -b issue1234
```

Commit the changes to your branch

```
git add .
git commit
```

> Before  sending the pull request, make sure your code is running on the latest available code by rebasing onto the upstream source

Reviewer: ```suggestion
Before sending the pull request, make sure your code is running on the latest available code by rebasing onto the upstream source
```

```
git fetch upstream
git rebase upstream/main
```

Verify your changes

```
npm run test
```

Push your changes

```
git push origin issue1234
```

Send the [pull request](https://docs.github.com/en/pull-requests), make requested changes, and get merged 

> 
> 
> 

Reviewer: ```suggestion
```

## Commit messages

* Limit the first line of the commit message (message summary) to 72 characters or less
* Start the commit message with an applicable tag:
    * `fix`  for a bug fix
    * `feat`  either for a backwards-compatible enhancement or for a rule change that adds reported problems
    * `fix!`  for a backwards-incompatible bug fix
    * `feat!` for a backwards-incompatible enhancement or feature
    * `docs`  for changes to documentation only
    * `chore` for changes that aren't user-facing
    * `build` for changes to build process only
    * `refactor` a change that doesn't affect APIs or user experience
    * `test` just changes to test files
    * `ci` changes to our CI configuration files and scripts

>     * `perf`  a code change that improves performance

Reviewer: these are angular's/conventional-commit's commit message conventions, and i don't use them. in particular, i'm philosophically opposed to "chore" - chores are bad, repo maintenance is good.

my commit conventions are undocumented but somewhat consistent. i suppose i should probably document them in a gist, and then we can link to that here.

Author: I've seen those listed on the ESLint website, this is why I've included them here😄. Will remove them for now and wait for your gist.

* Use the present tense and imperative mood when providing a description of what you did
* If your PR addresses an issue, reference it in the body of the commit message

#### Example

```
[tag]: Short description of what you did

Longer description here if necessary

Fixes #1234
```

## Code of Conduct

> We adhere to the [OpenJS Foundation Code of Conduct](https://github.com/openjs-foundation/cross-project-council/blob/main/CODE_OF_CONDUCT.md) and expect

Reviewer: this is a great idea, but we don't actually adhere to that in this github org yet. we could add a CODE_OF_CONDUCT.md file to https://github.com/jsx-eslint/.github, though, and then we don't need to mention it here at all.
